object_store/src/delimited.rs

```diff
@@ -125,8 +124,12 @@ impl LineDelimiter {
     /// Returns `true` if there is no remaining data to be read
     fn finish(&mut self) -> Result<bool> {
         if !self.remainder.is_empty() {
-            ensure!(!self.is_quote, UnterminatedStringSnafu);
-            ensure!(!self.is_quote, TrailingEscapeSnafu);
+            if self.is_quote {
+                return Err(Error::UnterminatedString.into());
+            }
+            if self.is_escape {
+                return Err(Error::TrailingEscape.into());
+            }
```

Comment:
```suggestion
            if self.is_quote {
                Err(Error::UnterminatedString)?;
            }
            if self.is_escape {
                Err(Error::TrailingEscape)?;
            }
```

Looks a bit simpler.